server/global/images_importation/ImagesList.js

```diff
@@ -0,0 +1,171 @@
+import { queryFormat, queryPromise } from "../../db/database.js";
+import { removeExtension } from "../files.js";
+import { AnalyzerWarning, getDuplicates } from "../importationUtils.js";
+import { Molecule } from "../molecules_importation/MoleculesList.js";
+
+const { normalizeDCI } = Molecule;
+
+const VALID_IMAGE_EXTENSIONS = ["jpeg", "jpg", "svg", "png"];
+const VALID_IMAGE_FORMAT_REGEX = new RegExp(`\\.${VALID_IMAGE_EXTENSIONS.join("|")}\\s*$`, "i");
+
+export const IMAGE_WARNINGS = {
+  INVALID_FORMAT: "INVALID_FORMAT",
+  INVALID_MOLECULE: "INVALID_MOLECULE",
+  DUPLICATED_IMAGES: "DUPLICATED_IMAGES",
+  UNKNOWN_MOLECULE: "UNKNOWN_MOLECULE",
+};
+
+/**
+ * Class representing a list of images
+ */
+export default class ImagesList {
+  /**
+   * Create an images list
+   * @param {string[]} imageNames
+   */
+  constructor(imageNames) {
+    this.list = imageNames.map((name) => new Image(name));
+  }
+
+  /**
+   * Analyze the images list
+   * @returns {AnalyzerWarning[]} A list of warnings
+   */
+  async analyze() {
+    const warnings = [];
+
+    const normalizedMoleculeNames = this.list.map((image) =>
+      removeExtension(normalizeDCI(image.name))
+    );
+
+    warnings.push(
+      ...getDuplicates(normalizedMoleculeNames).map(
+        (dup) =>
+          new AnalyzerWarning(
+            IMAGE_WARNINGS.DUPLICATED_IMAGES,
+            `Plusieurs images pour la molécule "${dup}"`
+          )
+      )
+    );
+
+    warnings.push(
+      ...(await this.getUnknownMolecules()).map(
+        (molecule) =>
+          new AnalyzerWarning(IMAGE_WARNINGS.UNKNOWN_MOLECULE, `Molécule inconnue : "${molecule}"`)
+      )
+    );
+
+    for (const image of this.list) {
+      warnings.push(...image.analyze());
+    }
+
+    return warnings;
+  }
+
+  /**
+   * Get all molecules from this list unknown in database
+   * @returns {string[]} The unknown molecule names
+   */
+  async getUnknownMolecules() {
+    const sql = `SELECT mo_dci FROM molecule;`;
+    const dbMolecules = (await queryPromise(sql)).map((molecule) => normalizeDCI(molecule.mo_dci));
```

Comment:
Is there any particular reason to put this in brackets?

Reply:
Brackets are needed to map the result of the promise once it's resolved.

Reply:
Oh okay, than you for the explanation 👍